### DISCUSSION TITLE
Pass HxMessageBox Settings (ButtonSettings ) to it?

### DISCUSSION THREAD

**Discussion by chobo2** [havit/Havit.Blazor#790] [Q&A]
Hi

I am using 

```
public class AppButtonSettings
{
	public static ButtonSettings MainButton { get; } = new()
	{
		Color = ThemeColor.Primary,
		Icon = BootstrapIcon.Fullscreen
	};

	public static ButtonSettings SecondaryButton { get; } = new()
	{
		Color = ThemeColor.Secondary,
		Outline = true
	};

	public static ButtonSettings CloseButton { get; } = new()
	{
		Color = ThemeColor.Danger,
		Outline = true,
		Icon = BootstrapIcon.XLg
	};
}
```

I want to send my customizations to the HxMessageBox to get the same styling across those buttons as well, but there seems to be no way to do this.

**Comment by crdo** [comment]
@chobo2 that is what `PrimaryButtonSettings` and `SecondaryButtonSettings` on `HxMessageBox` is for.

**Comment by chobo2** [comment]
> @chobo2 that is what `PrimaryButtonSettings` and `SecondaryButtonSettings` on `HxMessageBox` is for.

I did see that, but I am unclear where to set that.

**Comment by hakenr** [comment]
In you UI startup code, eg. `Program.cs`:
```csharp
HxMessageBox.Defaults.PrimaryButtonSettings = AppButtonSettings.MainButton;
HxMessageBox.Defaults.SecondaryButtonSettings = AppButtonSettings.SecondaryButton;
```

**Reply by chobo2** [reply]
Okay yeah I figured that out by downloading git repo. So the only way is to do it globally then. If for some reason I needed to change the button's color on some other page, is there away to override it in that case?
